2_intermediate/chapter10/solutions/img_avg.py

```diff
@@ -12,82 +12,92 @@
 import numpy
 import matplotlib.pyplot as plt
 
+# Code that grabs the image from the internet and makes it into an array
 url = "https://images.dog.ceo/breeds/waterdog-spanish/20180723_185544.jpg"
-img = numpy.array(Image.open(requests.get(url, stream=True).raw))
+img = numpy.array(Image.open(requests.get(url, stream=True).raw)).tolist()
 newimg = img
 transpose = numpy.transpose(img)
 
+plt.imshow(img)
+plt.show()
 
-# write code to create newimg here
-def solution1():
-    """
-    Iterating over the image here. i is a variable from
-    0 to the width of the image.
-    j is a variable that ranges from 0 to the height of the image.
-    i is associated with values
-    """
-    for i in range(len(img)):
-        for j in range(len(img[0])):
-            x_n = [0]
-            y_n = [0]
-
-            if i == 0:
-                x_n.append(1)
-            elif i == len(img) - 1:
-                x_n.append(-1)
-            else:
-                x_n.append(1)
-                x_n.append(-1)
-
-            if j == 0:
-                y_n.append(1)
-            elif j == len(img[0]) - 1:
-                y_n.append(-1)
-            else:
-                y_n.append(1)
-                y_n.append(-1)
-
-            r_avg = -1 * img[i][j][0]
-            g_avg = -1 * img[i][j][1]
-            b_avg = -1 * img[i][j][2]
-            c = -1
-
-            for x in x_n:
-                for y in y_n:
-                    r_avg += img[i + x][j + y][0]
-                    g_avg += img[i + x][j + y][1]
-                    b_avg += img[i + x][j + y][2]
-                    c += 1
-            r_avg = r_avg / c
-            g_avg = g_avg / c
-            b_avg = b_avg / c
-
-            newimg[i][j] = [r_avg, g_avg, b_avg]
-
-
-solution1()
 
+# Write code to create newimg here
+"""
+Iterating over the image here. i is a variable from
+0 to the width of the image.
+j is a variable that ranges from 0 to the height of the image.
+i is associated with values
+"""
+for i in range(len(img)):
+    for j in range(len(img[0])):
+        x_n = [0]
+        y_n = [0]
+
+        if i == 0:
+            x_n.append(1)
+        elif i == len(img) - 1:
+            x_n.append(-1)
+        else:
+            x_n.append(1)
+            x_n.append(-1)
+
+        if j == 0:
+            y_n.append(1)
+        elif j == len(img[0]) - 1:
+            y_n.append(-1)
+        else:
+            y_n.append(1)
+            y_n.append(-1)
+
+        r_avg = -1 * img[i][j][0]
+        g_avg = -1 * img[i][j][1]
+        b_avg = -1 * img[i][j][2]
+        c = -1
+
+        for x in x_n:
+            for y in y_n:
+                r_avg += img[i + x][j + y][0]
+                g_avg += img[i + x][j + y][1]
+                b_avg += img[i + x][j + y][2]
+                c += 1
+        r_avg = r_avg / c
+        g_avg = g_avg / c
+        b_avg = b_avg / c
+
+        newimg[i][j] = [r_avg, g_avg, b_avg]
+
+
+# Code that displays the new image at the end.
 plt.imshow(newimg)
```

Comment:
Okay so... I tried to run this solution code after I installed all the modules (yay for pip and pyenv), but I can only see the original image of the dog. I also got a `TypeError` which I'm not sure how to resolve because I don't really understand the code:
```
Traceback (most recent call last):
  File "/Users/rebeccadang/Desktop/Code/code4tomorrow/python/2_intermediate/chapter10/solutions/img_avg.py", line 115, in <module>
    plt.imshow(transpose)
  File "/Users/rebeccadang/.pyenv/versions/3.8.5/lib/python3.8/site-packages/matplotlib/pyplot.py", line 2707, in imshow
    __ret = gca().imshow(
  File "/Users/rebeccadang/.pyenv/versions/3.8.5/lib/python3.8/site-packages/matplotlib/__init__.py", line 1431, in inner
    return func(ax, *map(sanitize_sequence, args), **kwargs)
  File "/Users/rebeccadang/.pyenv/versions/3.8.5/lib/python3.8/site-packages/matplotlib/axes/_axes.py", line 5519, in imshow
    im.set_data(X)
  File "/Users/rebeccadang/.pyenv/versions/3.8.5/lib/python3.8/site-packages/matplotlib/image.py", line 706, in set_data
    raise TypeError("Invalid shape {} for image data"
TypeError: Invalid shape (3, 1008, 490) for image data
```

Reply:
Outdated comment.